Libraries/pbxbuild/Sources/Target/Environment.cpp

```diff
@@ -384,13 +384,21 @@ Create(Build::Environment const &buildEnvironment, Build::Context const &buildCo
         pbxsetting::Setting::Parse("GCC_VERSION", "$(DEFAULT_COMPILER)"),
     }), false);
 
-    environment.insertFront(sdk->platform()->defaultProperties(), false);
+    if (sdk->platform()->defaultProperties()) {
+        environment.insertFront(*sdk->platform()->defaultProperties(), false);
+    }
     environment.insertFront(PlatformArchitecturesLevel(buildEnvironment.specManager(), specDomains), false);
-    environment.insertFront(sdk->defaultProperties(), false);
+    if (sdk->defaultProperties()) {
+        environment.insertFront(*sdk->defaultProperties(), false);
+    }
     environment.insertFront(sdk->platform()->settings(), false);
```

Comment:
With this change, `defaultProperties()` becomes an `ext::optional<pbxsetting::Level>`, which means it may not exist. (Depending on whether it existed in the original file.) `settings()` is not optional, since it's synthesized at runtime, so nothing to check for.

The advantage of storing optionals is that it maintains the information of whether the source file had an empty `defaultProperties()` (optional has value, value is empty), or didn't have one at all (optional has no value).